shared-test/jwt-utils.ts

```diff
@@ -1,18 +1,66 @@
-import { base64url, JWTHeaderParameters, JWTPayload } from "jose";
-import { randomString } from "./string-utils";
-import { JWTClass } from "@govuk-one-login/data-vocab/credentials";
+import { DIDDocument } from "did-resolver";
+import { CompactJWSHeaderParameters, CompactSign, importJWK, JWK, JWTHeaderParameters } from "jose";
+import { signKms } from "../tests/acceptance/steps/utils/kms-utils";
 
-export const sign = (jwtHeader: JWTHeaderParameters, jwtPayload: JWTPayload | JWTClass): string => {
-  const header = base64url.encode(Buffer.from(JSON.stringify(jwtHeader)));
-  const payload = base64url.encode(Buffer.from(JSON.stringify(jwtPayload)));
-  const signature = base64url.encode(Buffer.from(randomString(16)));
-  return `${header}.${payload}.${signature}`;
+export const privateKeyJwk: JWK = {
+  kty: "EC",
+  crv: "P-256",
+  x: "f83OJ3D2xF1Bg8vub9tLe1gHMzV76e8Tus9uPHvRVEU", // pragma: allowlist secret
+  y: "x_FEzRu9m36HLN_tue659LNpXW6pCyStikYjKIWI5a0", // pragma: allowlist secret
+  d: "jpsQnnGQmL-YBIffH1136cspYG6-0iY7X1fCE9-E9LI", // pragma: allowlist secret
 };
 
-export function getDefaultJwtHeader(alg: string = "EC"): JWTHeaderParameters {
+export const publicKeyJwk: JWK = {
+  kty: "EC",
+  crv: "P-256",
+  x: privateKeyJwk.x,
+  y: privateKeyJwk.y,
+};
+
+export const did = "did:web:api.identity.dev.account.gov.uk";
+export const verificationMethodId = `${did}#f5fe5d2a-9eb6-4819-8c46-723e3a21565a`;
+
+export const didDocument: DIDDocument = {
+  "@context": ["https://www.w3.org/ns/did/v1", "https://w3id.org/security/jws/v1"],
+  id: did,
+  controller: did,
+  assertionMethod: [
+    {
+      id: verificationMethodId,
+      type: "JsonWebKey2020",
+      controller: did,
+      publicKeyJwk,
+    },
+  ],
+};
+
+export async function sign(
+  header: Record<string, any>,
+  body: Record<string, any>,
+  kms: boolean = false
+): Promise<string> {
+  const protectedHeader: CompactJWSHeaderParameters = {
+    typ: header?.typ ?? "JWT",
+    kid: header?.kid ?? verificationMethodId,
+    alg: header?.alg ?? "ES256",
+    ...header,
+  };
+  return kms ? await signKms(body, protectedHeader) : await signLocal(body, protectedHeader);
+}
+
+async function signLocal(body: Record<string, any>, protectedHeader: CompactJWSHeaderParameters) {
+  const key = await importJWK(privateKeyJwk, "ES256");
+  const payloadBytes = new TextEncoder().encode(JSON.stringify(body));
+  return await new CompactSign(payloadBytes).setProtectedHeader(protectedHeader).sign(key);
+}
+
+export function getDefaultStoredIdentityHeader(
```

Comment:
This was named `getDefaultJwtHeader` because it was used as the header for both Identity Check Verifiable Credentials and for Stored Identities. So it was generic. 

This is now specific to stored identity header but still used in places as the header for VCs and it doesn't really make sense, because the `did` kid is specific to stored identities. VCs will use non did stuff.

Maybe for a follow up tidy up PR, but please can we split this out so that it makes more sense for the two types of headers.